src/config_generator/from_proto.rs

```diff
@@ -0,0 +1,441 @@
+        proto_path.push("proto");
+        proto_path.push(name);
+        Ok(protox_parse::parse(
+            name,
+            std::fs::read_to_string(proto_path)?.as_str(),
+        )?)
+    }
+
+    #[test]
+    fn test_from_proto() -> anyhow::Result<()> {
+        // news_enum.proto covers:
+        // test for mutation
+        // test for empty objects
+        // test for optional type
+        // test for enum
+        // test for repeated fields
+        // test for a type used as both input and output
+        // test for two types having same name in different packages
+
+        let mut set = FileDescriptorSet::default();
+
+        let news = get_proto_file_descriptor("news.proto")?;
+        let greetings_a = get_proto_file_descriptor("greetings_a.proto")?;
+        let greetings_b = get_proto_file_descriptor("greetings_b.proto")?;
+
+        set.file.push(news);
+        set.file.push(greetings_a);
+        set.file.push(greetings_b);
+
+        let result = from_proto(vec![set], "Query").to_sdl();
+
+        insta::assert_snapshot!(result);
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_config_from_sdl() -> anyhow::Result<()> {
+        let mut set = FileDescriptorSet::default();
+
+        let news = get_proto_file_descriptor("news.proto")?;
+        let greetings_a = get_proto_file_descriptor("greetings_a.proto")?;
+        let greetings_b = get_proto_file_descriptor("greetings_b.proto")?;
+
+        set.file.push(news.clone());
+        set.file.push(greetings_a.clone());
+        set.file.push(greetings_b.clone());
+
+        let result = from_proto(vec![set], "Query").to_sdl();
+
+        // test for different sets
+        let mut set = FileDescriptorSet::default();
+        let mut set1 = FileDescriptorSet::default();
+        let mut set2 = FileDescriptorSet::default();
+        set.file.push(news);
+        set1.file.push(greetings_a);
+        set2.file.push(greetings_b);
+
+        let result_sets = from_proto(vec![set, set1, set2], "Query").to_sdl();
+
+        pretty_assertions::assert_eq!(result, result_sets);
+        Ok(())
+    }
+
+    #[test]
+    fn test_required_types() -> anyhow::Result<()> {
+        // required fields are deprecated in proto3 (https://protobuf.dev/programming-guides/dos-donts/#add-required)
+        // this example uses proto2 to test the same.
+        // for proto3 it's guaranteed to have a default value (https://protobuf.dev/programming-guides/proto3/#default)
+        // and our implementation (https://github.com/tailcallhq/tailcall/pull/1537) supports default values by default.
+        // so we do not need to explicitly mark fields as required.
+
+        let mut set = FileDescriptorSet::default();
+        let req_proto = get_proto_file_descriptor("person.proto")?;
+        set.file.push(req_proto);
+
+        let cfg = from_proto(vec![set], "Query").to_sdl();
+        insta::assert_snapshot!(cfg);
+
+        Ok(())
+    }
+    #[test]
+    fn test_get_value() {
```

Comment:
Please split this into individual asserts. We know exactly which test is passing and which ones are failing that way.